src/simplest-rust/src/lib.rs

```diff
@@ -0,0 +1,120 @@
+use byteorder::{ByteOrder, LittleEndian};
+use solana_sdk::{
+    account_info::AccountInfo, entrypoint, entrypoint::ProgramResult, info,
+    program_error::ProgramError, account_info::next_account_info, pubkey::Pubkey,
+};
+use std::mem;
+
+// Declare and export the program's entrypoint
+entrypoint!(process_instruction);
+
+// Program entrypoint's implementation
+fn process_instruction<'a>(
+    program_id: &Pubkey,             // Public key of program account
+    accounts: &'a [AccountInfo<'a>], // data accounts
+    instruction_data: &[u8],         // 1 = vote for A, 2 = vote for B
+) -> ProgramResult {
+
+    info!("Rust program entrypoint");
+
+    // Iterating accounts is safer then indexing
+    let accounts_iter = &mut accounts.iter();
+
+    // Get the account that holds the vote count
+    let account = next_account_info(accounts_iter)?;
+
+    // The account must be owned by the program in order to modify its data
+    if account.owner != program_id {
+        info!("Vote account does not have the correct program id");
+        return Err(ProgramError::IncorrectProgramId);
+    }
+
+    // The data must be large enough to hold two u32 vote counts
+    if account.try_data_len()? < 2 * mem::size_of::<u32>() {
```

Comment:
determination of valid account data length could be simplified by implementing `solana_sdk::program_pack::Pack`